src/routes/blog/_query.ts

```diff
@@ -103,6 +113,14 @@ export async function fetchNewsletters(): Promise<Newsletter[]> {
     body: JSON.stringify({ query: newslettersQuery }),
   });
 
-  const newsletters = formatNewsletters(await response.json());
-  return newsletters;
+  let newsletters = formatNewsletters(await response.json());
+  const tags = discernTags(newsletters);
+
+  if (options && options.labels.length > 0) {
```

Comment:
Nit: Strictly assert that the `options` object is defined.
```suggestion
  if (options !== undefined && options.labels.length > 0) {
```

Reply:
Alternatively, just `if (options?.labels.length > 0)`

Reply:
More info: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

> The ?. operator is like the . chaining operator, except that instead of causing an error if a reference is [nullish](https://developer.mozilla.org/en-US/docs/Glossary/Nullish) ([null](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/null) or [undefined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined)), the expression short-circuits with a return value of undefined. When used with function calls, it returns undefined if the given function does not exist.

Reply:
Thank you @EthanThatOneKid for pointing out. 
I like your solution @jaasonw. Tested and it worked as expected. Thanks